lib/api/apiUtils/bucket/bucketWebsite.js

```diff
@@ -0,0 +1,275 @@
+        errMsg = 'Invalid website configuration xml';
+        return errors.MalformedXML.customizeDescription(errMsg);
+    }
+
+    const resultConfig = jsonResult.WebsiteConfiguration;
+    if (!resultConfig.IndexDocument && !resultConfig.RedirectAllRequestsTo) {
+        errMsg = 'Value for IndexDocument Suffix must be provided if ' +
+        'RedirectAllRequestsTo is empty';
+        return errors.InvalidArgument.customizeDescription(errMsg);
+    }
+
+    if (resultConfig.RedirectAllRequestsTo) {
+        const parent = resultConfig.RedirectAllRequestsTo;
+        const redirectAllObj = {};
+        if (resultConfig.IndexDocument || resultConfig.ErrorDocument ||
+        resultConfig.RoutingRules) {
+            errMsg = 'RedirectAllRequestsTo cannot be provided in ' +
+            'conjunction with other Routing Rules.';
+            return errors.InvalidRequest.customizeDescription(errMsg);
+        }
+        if (!_xmlContainsElem(parent, 'HostName')) {
+            errMsg = 'RedirectAllRequestsTo not well-formed';
+            return errors.MalformedXML.customizeDescription(errMsg);
+        }
+        if (typeof parent[0].HostName[0] !== 'string'
+        || parent[0].HostName[0] === '') {
+            errMsg = 'Valid HostName required in RedirectAllRequestsTo';
+            return errors.InvalidRequest.customizeDescription(errMsg);
+        }
+        redirectAllObj.hostName = parent[0].HostName[0];
+        if (_xmlContainsElem(parent, 'Protocol')) {
+            if (parent[0].Protocol[0] !== 'http' &&
+            parent[0].Protocol[0] !== 'https') {
+                errMsg = 'Invalid protocol, protocol can be http or https. ' +
+                'If not defined, the protocol will be selected automatically.';
+                return errors.InvalidRequest.customizeDescription(errMsg);
+            }
+            redirectAllObj.protocol = parent[0].Protocol[0];
+        }
+        websiteConfig.setRedirectAllRequestsTo(redirectAllObj);
+    }
+
+    if (resultConfig.IndexDocument) {
+        const parent = resultConfig.IndexDocument;
+        if (!_xmlContainsElem(parent, 'Suffix')) {
+            errMsg = 'IndexDocument is not well-formed';
+            return errors.MalformedXML.customizeDescription(errMsg);
+        } else if (typeof parent[0].Suffix[0] !== 'string'
+        || parent[0].Suffix[0] === ''
+        || parent[0].Suffix[0].indexOf('/') !== -1) {
+            errMsg = 'IndexDocument Suffix is not well-formed';
+            return errors.InvalidArgument.customizeDescription(errMsg);
+        }
+        websiteConfig.setIndexDocument(parent[0].Suffix[0]);
+    }
+
+    if (resultConfig.ErrorDocument) {
+        const parent = resultConfig.ErrorDocument;
+        if (!_xmlContainsElem(parent, 'Key')) {
+            errMsg = 'ErrorDocument is not well-formed';
+            return errors.MalformedXML.customizeDescription(errMsg);
+        }
+        if (typeof parent[0].Key[0] !== 'string'
```

Comment:
I see this type of check `typeof foo !== 'string' || foo === ''` occurring frequently. Perhaps you could extract it into some find of utility check function that returns a boolean.